fluentd/configs.d/filter/k8s_record_transform.conf

```diff
@@ -3,7 +3,7 @@
   enable_ruby
   <record>
     hostname ${(kubernetes_host rescue nil) || File.open('/etc/docker-hostname') { |f| f.readline }.rstrip}
-    message ${log}
+    message ${(message rescue nil) || log}
     version 1.1.4
   </record>
   remove_keys log,stream
```

Comment:
I don't think there are any *common* conditions where you'll have both. You
can put any metadata in that you want with JSON logging though, and that's
just the kind of thing that spurred this issue.

I'm not sure this DSL is adequately expressive for the logic you'd want.

On Wed, May 11, 2016 at 4:09 PM, Richard Megginson <notifications@github.com
> wrote:

> In fluentd/configs.d/filter/k8s_record_transform.conf
> <https://github.com/openshift/origin-aggregated-logging/pull/131#discussion_r62917418>
> :
>
> > @@ -3,7 +3,7 @@
> >    enable_ruby
> >    <record>
> >      hostname ${(kubernetes_host rescue nil) || File.open('/etc/docker-hostname') { |f| f.readline }.rstrip}
> > -    message ${log}
> > +    message ${(message rescue nil) || log}
> >      version 1.1.4
> >    </record>
> >    remove_keys log,stream
>
> Under what conditions will a message have both message and log? Can you
> provide an example?
> Perhaps compare message and log - if they are the same, just keep message,
> otherwise keep both. Also note that this may add to the amount of data per
> log entry in ES, if it is a frequent occurrence that an entry has both log
> and message and they are different.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin-aggregated-logging/pull/131/files/ba6b2abe01fb8949114a4b852d502cf9863d7913#r62917418>
>